src/test/java/com/google/starfish/services/FavoriteNoteServiceTest.java

```diff
@@ -0,0 +1,300 @@
+package com.google.starfish.services;
+
+import java.util.Date;
+import java.util.Calendar;
+import java.sql.SQLException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import javax.sql.DataSource;
+import static com.ninja_squad.dbsetup.Operations.*;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+import com.google.starfish.services.FavoriteNoteService.Recency;
+import com.google.starfish.models.Note;
+import com.ninja_squad.dbsetup.DbSetup;
+import com.ninja_squad.dbsetup.destination.DataSourceDestination;
+import com.ninja_squad.dbsetup.operation.Operation;
+
+@RunWith(JUnit4.class)
+public class FavoriteNoteServiceTest {
+  private Constants constants = new Constants();
+  private String NOTES = constants.NOTES;
+  private String LABELS = constants.LABELS;
+  private String FAVORITE_NOTES = constants.FAVORITE_NOTES;
+  private String USERS = constants.USERS;
+  private DataSource pool = Constants.pool;
+  private FavoriteNoteService favoriteNoteService = new FavoriteNoteService();
+
+  private final Date today = favoriteNoteService.getDateBasedOnRecency(Recency.TODAY);
+  private final Date thisWeek = favoriteNoteService.getDateBasedOnRecency(Recency.THIS_WEEK);
+  private final Date thisMonth = favoriteNoteService.getDateBasedOnRecency(Recency.THIS_MONTH);
+  private final Date allTime = favoriteNoteService.getDateBasedOnRecency(Recency.ALL_TIME);
+
+  private final String newSchool = "cornell university";
+  private final String newCourse = "cs1300";
+  private final String firstNewUserId = Integer.toString(Integer.parseInt(constants.REFERENCE_USER_ID) + 1);
+  private final String secondNewUserId = Integer.toString(Integer.parseInt(constants.REFERENCE_USER_ID) + 2);
+
+  private static final boolean runTests = Constants.TEST_DB_NAME.equals("starfish_test");
+
+  /** Clear DB and insert reference data */
+  @Before
+  public void prepare() throws Exception {
+    if(!runTests) throw new Exception("Wrong Test Database Name");
+    Operation operation = getBeforeTestOperation();
+    DbSetup dbSetup = new DbSetup(new DataSourceDestination(pool), operation);
+    dbSetup.launch();
+  }
+
+  /** Test to see if we can get all notes favorited by a single user */
+  @Test
+  public void testGettingFavoriteNotesByUserId() throws SQLException {
```

Comment:
Ideally, this test would actually verify that this returns the entire array of notes as you would expect to receive, not just the length (that can perhaps just be a test to preface checking the entire value because if that's not true, then you needn't bother checking for the values).

Reply:
Ahh this makes sense, the test does definitely feel incomplete without checking the actual values of the notes, even if the test passes, there's a lingering feeling of "what if the notes retrieved were wrong?" I'll make a change for this!